metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java

```diff
@@ -159,4 +256,226 @@ private void publishToSolr(RdfWrapper rdf, FullBeanImpl fullBean) throws Indexin
+    if (!CollectionUtils.isEmpty(identifiers)) {
+      if (!CollectionUtils.isEmpty(titles)) {
+        listsToCombineMaps.put(EdmLabel.PROXY_DC_IDENTIFIER.toString(), identifiers);
+        listsToCombineMaps.put(EdmLabel.PROXY_DC_TITLE.toString(), titles);
+      } else if (!CollectionUtils.isEmpty(descriptions)) {
+        listsToCombineMaps.put(EdmLabel.PROXY_DC_IDENTIFIER.toString(), identifiers);
+        listsToCombineMaps.put(EdmLabel.PROXY_DC_DESCRIPTION.toString(), descriptions);
+      }
+    }
+    return listsToCombineMaps;
+  }
+
+  private HashMap<String, List<String>> createSecondCombinationGroup(List<String> isShownByList,
+      List<String> titles, List<String> descriptions) {
+    HashMap<String, List<String>> listsToCombineMaps = new HashMap<>();
+    if (!CollectionUtils.isEmpty(isShownByList)) {
+      if (!CollectionUtils.isEmpty(titles)) {
+        listsToCombineMaps
+            .put(EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_BY.toString(), isShownByList);
+        listsToCombineMaps.put(EdmLabel.PROXY_DC_TITLE.toString(), titles);
+      } else if (!CollectionUtils.isEmpty(descriptions)) {
+        listsToCombineMaps
+            .put(EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_BY.toString(), isShownByList);
+        listsToCombineMaps.put(EdmLabel.PROXY_DC_DESCRIPTION.toString(), descriptions);
+      }
+    }
+    return listsToCombineMaps;
+  }
+
+  private HashMap<String, List<String>> createThirdCombinationGroup(List<String> isShownByList,
+      List<String> identifiers) {
+    HashMap<String, List<String>> listsToCombineMaps = new HashMap<>();
+    if (!CollectionUtils.isEmpty(isShownByList) && !CollectionUtils.isEmpty(identifiers)) {
+      listsToCombineMaps
+          .put(EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_BY.toString(), isShownByList);
+      listsToCombineMaps.put(EdmLabel.PROXY_DC_IDENTIFIER.toString(), identifiers);
+    }
+    return listsToCombineMaps;
+  }
+
+  private HashMap<String, List<String>> createFourthGroup(List<String> isShownByList,
+      List<String> isShownAtList) {
+    final HashMap<String, List<String>> listsToCombineMapsIsShownAtAndBy = new HashMap<>();
+    if (!CollectionUtils.isEmpty(isShownAtList) && !CollectionUtils.isEmpty(isShownByList)) {
+      listsToCombineMapsIsShownAtAndBy
+          .put(EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_BY.toString(), isShownByList);
+      listsToCombineMapsIsShownAtAndBy
+          .put(EdmLabel.PROVIDER_AGGREGATION_EDM_IS_SHOWN_AT.toString(), isShownAtList);
+    }
+    return listsToCombineMapsIsShownAtAndBy;
+  }
+
+  private String generateQueryForFields(Map<String, List<String>> listsToCombine) {
+    final String combinationResult = listsToCombine.entrySet().stream()
+        .map(entry -> generateOrOperationFromList(entry.getKey(), entry.getValue()))
+        .collect(Collectors.joining(" AND "));
+    return String.format("(%s)", combinationResult);
```

Comment:
There is a Collectors.joining method that accepts a prefix and suffix. So no need to do the subsequent String.format.

Reply:
Updated